docs/default.nix

```diff
@@ -37,11 +37,19 @@ let
       inherit lib;
 
       nixos-render-docs = prev.nixos-render-docs.overrideAttrs (old: {
+        propagatedBuildInputs = old.propagatedBuildInputs ++ [
```

Comment:
I know this is kinda off-topic, but I can never wrap my head around when to use native or not and propagated or not for buildInputs...

My understanding is:
- buildInputs generally only relate to the drv builder, not the package's runtime
- `native` buildInputs are for exes that should be run from the PATH during the build
- `propagated` buildInputs are for deps that should also be available to builders of _other packages_ that include this package in _their_ buildInputs (i.e. transitives get propagated)
- propagatedNativeBuildInputs is also a thing, see above two

Here we shouldn't need propagated right?

Reply:
> I know this is kinda off-topic, but I can never wrap my head around when to use native or not and propagated or not for buildInputs...
> 
> My understanding is:
> 
>     * buildInputs generally only relate to the drv builder, not the package's runtime
> 
>     * `native` buildInputs are for exes that should be run from the PATH during the build
> 
>     * `propagated` buildInputs are for deps that should also be available to builders of _other packages_ that include this package in _their_ buildInputs (i.e. transitives get propagated)
> 
>     * propagatedNativeBuildInputs is also a thing, see above two
> 
> 
> Here we shouldn't need propagated right?

buildInputs are runtime available packages, actually.
native are build only
propogated do the same but to derivations that inherit it.